presto-parser/src/main/java/com/facebook/presto/sql/tree/GroupingElement.java

```diff
@@ -31,4 +31,16 @@ protected GroupingElement(Optional<NodeLocation> location)
     {
         return visitor.visitGroupingElement(this, context);
     }
+
+    void validateExpressions(List<Expression> expressions)
+    {
+        expressions.forEach(expression -> ExpressionTreeRewriter.rewriteWith(new ExpressionRewriter<Void>()
+        {
+            @Override
+            public Expression rewriteLambdaExpression(LambdaExpression node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+            {
+                throw new UnsupportedOperationException("GROUP BY does not support lambda expressions, please use GROUP BY # instead");
```

Comment:
nit: just curious if user understand what does `GROUP BY #` mean :)

Reply:
Suggestions on better error message are welcomed. I don't really know what this should be referred to as. @kaikalur